### ISSUE TITLE
Incorrect warning for `workflow_dispatch` input type `number`.

### ISSUE THREAD

**Issue by inahga** [rhysd/actionlint#319]
Example workflow:
```yaml
name: test

on:
  workflow_dispatch:
    inputs:
      test:
        type: number
        default: "0"

jobs:
  job1:
    runs-on: ubuntu-latest
    steps:
      - run: echo hello world
```

I receive this lint:
```
test.yml:7:15: input type of workflow_dispatch event must be one of "string", "boolean", "choice", "environment" but got "number" [syntax-check]
  |
7 |         type: number
  |               ^~~~~~
```

Actionlint version:
```
🐧 actionlint --version
v1.6.25
installed by building from source
built with go1.20.4 compiler for linux/amd64
```

This is not correct, `number` is valid for the input type. See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputsinput_idtype. Github also accepts the input just fine.

(more than happy to work on this myself, just filing an issue so I don't forget!)

**Comment by inahga** [comment]
Whoops, dupe of https://github.com/rhysd/actionlint/issues/316. I just need to wait for the next release :)
